### ISSUE TITLE
Bug: the type of ReactDOM.prefetchDNS does not exist.

### ISSUE THREAD

**Issue by hmartiins** [facebook/react#27170]
React version: 18.2.0

## Steps To Reproduce

1. Import ReactDOM
2. Create a simple component
3. Add 
```js
 ReactDOM.prefetchDNS('...')
```
4. Import this component into your global App.
5. When inspecting the page, you will see that "dns-prefetch" tags have been added with the given link.

Link to code example: https://codesandbox.io/p/sandbox/keen-drake-tq2t29

![image](https://github.com/facebook/react/assets/51277667/3721b096-c6ee-4c10-86cb-acc1ab744607)
If we inspect the page, we verify that the link with dns-prefetch, was inserted correctly. Even ReactDOM.prefetchDNS method showing error that prefetchDNS does not exist.

## The current behavior

When trying to use ``ReactDOM.prefetchDNS`` we get a type error.
```
Property 'prefetchDNS' does not exist on type 'typeof import(".../node_modules/@types/react-dom/index")'.ts(2339)
```

## The expected behavior
We shouldn't have this typo, since the method exists and is working.

**Comment by Gauravpadam** [comment]
I can second to the error you've received , prefetchDNS is _indeed_ not found in 'react-dom'

![image](https://github.com/facebook/react/assets/12787482/1f253035-fd4b-4621-b8fc-b523ac1c3505)

Closest I've ever come across is `ReactDOM.prefetch(url)`

However I'm not able to produce the same skeleton result as you

```
import React from "react";
import ReactDOM from "react-dom";

export function PreloadResources() {
  ReactDOM.prefetchDNS("https://www.google.com/");

  return null;
}


function App() {
  return (
    <>
      <PreloadResources />
    </>
  );
}

export default App;
```

It's a small component, similar to yours
maybe its the code that is incorrect?
other devs can take a look

**Comment by hmartiins** [comment]
@Gauravpadam the difference I saw is that I call my component within the head tag of my site, in your code you are calling it in the body of the page, and it would be cool to see if the react-dom is up to date, as these functions are still very recent

**Comment by hmartiins** [comment]
From what I've seen the types are even implemented in the [ReactDOMTypes](https://github.com/facebook/react/blob/493f72b0a7111b601c16b8ad8bc2649d82c184a0/packages/react-dom/src/shared/ReactDOMTypes.js#L32). But it is not up to date with the @types/react-dom type library, we can check it in [this link](
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-dom/canary.d.ts).

I can contribute by updating the type library, but I don't know if it's the right way to go..

**Comment by ssaurexd** [comment]
I solved this problem deleting node_modules,  package-lock.json and doing again `npm install`

**Comment by jadurani** [comment]
suggestion from the NextJS team:

> You can use // @ts-ignore as a temporary solution to avoid type errors.

<img width="653" alt="image" src="https://github.com/facebook/react/assets/11599005/1109ae9e-b54e-4701-94b0-cadc27d86ac0">

source: https://nextjs.org/docs/app/api-reference/functions/generate-metadata#link-reldns-prefetch

**Comment by hmartiins** [comment]
Yes, the functionality exists, but it has not been documented with types yet.

**Comment by eps1lon** [comment]
Fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/66896. Available as of `@types/react-dom@18.2.10`
